meta-resin-common/recipes-containers/docker/docker/docker.service

```diff
@@ -1,8 +1,8 @@
 [Unit]
 Description=Docker Application Container Engine
 Documentation=http://docs.docker.com
-After=network.target docker.socket
-Requires=docker.socket
+After=network.target docker.socket var-lib-docker.mount
+Requires=docker.socket var-lib-docker.mount
 
```

Comment:
Redo this commit to encompass all the changes to the systemd service.
Use the existing rce.service as a template.
First thing that comes into mind is that it looks like you are missing the rce.mount dependency entirely